core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/posix.c

```diff
@@ -1356,8 +1356,9 @@ wasmtime_ssp_fd_allocate(
     // conditions. We may end up shrinking the file right now.
     struct stat sb;
     int ret = fstat(fd_number(fo), &sb);
-    if (ret == 0 && sb.st_size < offset + len)
-        ret = ftruncate(fd_number(fo), offset + len);
+    off_t newsize = offset + len;
```

Comment:
How about changing to `off_t newsize = (off_t)(offset + len)`? So as to yield warning for some compilers?

Reply:
do you mean you prefer to have an explicit cast?

Reply:
Yes, not sure whether some compilers will report warning or not if not adding explicit cast.

Reply:
done